.circleci/config.yml

```diff
@@ -83,13 +91,54 @@ jobs:
           command: |
             docker push circleci/circleci-inferconfig:0.$CIRCLE_BUILD_NUM
             docker push circleci/circleci-inferconfig:latest
+  vulnerability-scan:
+    executor: go
+    steps:
+      - checkout
+      - run:
+          name: Setup Scanning
+          command: |
+              git config --global url."https://$GITHUB_USER:$GITHUB_TOKEN@github.com/circleci/".insteadOf "https://github.com/circleci/"
+      - when:
+          condition:
+              or:
+                - equal: [ main, << pipeline.git.branch >> ]
+          steps:
+              - run:
+                  name: Launching Snyk Orb Scanning
+                  command: echo "Running snyk/scan on main; uploading the results"
+              - run:
+                  name: Cleanup RemoteRepoURL
+                  command: echo 'export REMOTE_REPO_URL="${CIRCLE_REPOSITORY_URL%".git"}"' >> "$BASH_ENV"
+              - snyk/scan:
+                  organization: "circleci-public"
+                  fail-on-issues: true
+                  severity-threshold: high
+                  monitor-on-build: true
+                  additional-arguments: "--all-projects --remote-repo-url=${REMOTE_REPO_URL} -d"
+      - unless:
+          condition:
+              or:
+                - equal: [ main, << pipeline.git.branch >> ]
+          steps:
+              - run:
+                  name: Launching Snyk Orb Scanning
+                  command: echo "Running snyk/scan on branch; not uploading the results"
+              - snyk/scan:
+                  organization: "circleci-public"
+                  fail-on-issues: true
+                  severity-threshold: high
+                  monitor-on-build: false
+                  additional-arguments: "--all-projects -d"
 
 workflows:
   test-and-release:
     jobs:
       - test
       - infer-own-config
       - validate-inferred-configs
+      - vulnerability-scan:
+          context: org-global-employees
       - build-and-push-docker-image:
           requires:
```

Comment:
Do we want to still push docker images if the scan job fails?

Reply:
No. I'll make the `build-and-push-docker-image` job require the `vulnerability-scan` job.